### ISSUE TITLE
How do I destroy a TaxonName? (new feedback and functionality)

### ISSUE THREAD

**Issue by mjy** [SpeciesFileGroup/taxonworks#1554]
When I need to destroy a TaxonName, then I need to know how to do it, and when it fails to destroy I need to know the reasons why.

TaxonNames are highly constrained in the system, and must have no Core or Related data attached to them to be destroyable (Annotations will be destroyed automatically).

As a proxy to understand what is related to the name use the Radial navigator to move to TaxonName->Show, then use Related->Related data to see what is related to that name.  This will at least give you a list of things that must be be destroyed.

As a user I need a more concise list of blockers, and potentially the ability to destroy related data along with the name.  

* [ ] Add/improve error message feedback to destroy button in general

Perhaps also a destroy button with a modal that has

## Feedback
* [ ] Lists blocking relationships
* [ ] Lists counts of annotations that will be destroyed with the click

## Functions
* [ ] Ultimately provides an option to move all related data to another name (merge)
* [ ] Provides the option to destroy the name outright if there are no related data

See also Merge taxon name issues.

**Comment by proceps** [comment]
The best solution to destroy a TN, in my opinion, is to Merge one TN into another. There are only two scenarios when I needed to destroy a TN, I just created it, realized a mistake. Destroying is relatively easy. Second scenario, 90% of the cases, when I found a duplicate. And this could be painful. Find all TN relationships and reassign them to another taxon. Destroy an OTU.

I would propose to offer Merge functionality, when the user can decide what to move. 
1. TN relationships will be moved from one TN to another
2. TN classificaltions - optional
3. Authors - optional
4. Citations - optional.
5. All annotations - optional.
6. OTU - either assign to second TN, or merge
